cpp/lib/src/IssnLookup.cc

```diff
@@ -35,16 +35,29 @@ void ExtractingData(const std::string &issn, const nlohmann::json &issn_info_jso
                 if (ar.at("@id") == issn_uri) {
                     issn_info->issn_ = issn;
                     for (auto &[key, val] : ar.items()) {
-                        if (key == "mainTitle")
-                            issn_info->main_title_ = val;
+                        if (key == "mainTitle") {
+                            if (val.is_structured())
+                                for (auto val_item : val)
```

Comment:
Usually in all for-loops we should use references as well as const whenever possible
`for (const auto &val_item : val)`